dwm1001/examples/lis2dh12.rs

```diff
@@ -1,43 +1,77 @@
 //! Accesses the LIS2DH12 3-axis accelerometer
+//!
+//! To build:
+//! cargo build --example lis2dh12 --features "dev rt semihosting" --release
+//!
+//! To debug:
+//! openocd -f openocd_jlink.cfg
```

Comment:
I don't understand this. The `openocd.cfg` already uses JLink, so what does this one do differently?

Generally speaking, if there are other configurations that you find useful, I don't mind adding them to the repository, in addition to the current one.

Reply:
I added the vscode launch profile, that integrates with the `cortex-debug` plugin. (You can do even more there, like building before launching etc. but this was just a simple config.) I added an `vscode_openocd.cfg` file as well, to keep your `openocd.cfg` intact. Now I see why I needed my "own! openocd script, its since yours sets the `flash bank` and that is now comes built in with recent `openocd` distros.

An another note: I added documentation for the driver to your `src/lib.rs` but then it seems that `cargo fmt` changes a lot. I don't know if you want it hand formatted, to look nicer, if so there should be an attribute in the file, else it will break again (or put a custom `rustfmt.toml` in the repo), if you want control over the formatting.

Reply:
I'm on latest stable, maybe your code was formatted under nightly? (Jorge mentioned it could be that..)

Reply:
> Now I see why I needed my "own! openocd script, its since yours sets the `flash bank` and that is now comes built in with recent `openocd` distros.

Yeah, the configuration in this repo is written against the latest release of OpenOCD, which is years old. No surprise that other software that bundles OpenOCD uses newer versions.

> An another note: I added documentation for the driver to your `src/lib.rs` but then it seems that `cargo fmt` changes a lot. I don't know if you want it hand formatted, to look nicer, if so there should be an attribute in the file, else it will break again (or put a custom `rustfmt.toml` in the repo), if you want control over the formatting.

I'm not using `cargo fmt`, not because of any specific formatting decisions[1], but because I think that formatting code consistently is a flawed concept. Consistent formatting has its merits in certain scenarios, but the idea that consistent formatting should be applied universally is plain wrong.

This is the subject of a whole article I'm going to write one day, but here's the short version: I don't care about consistency, I care about readability. Consistency often helps readability, but in some cases is actively hinders it. One example for this (among many) is the usage of svd2rust-generated APIs, and since that example alone is highly relevant to my work, I don't use `cargo fmt`.

I generally don't care about the formatting of new code (as long as it's readable), but I don't want to merge a commit that re-formats most of the crate for no good reason.

[1] Although I do think the standard formatting is less readable than mine, but that's subjective, and something I could learn to live with.